utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/ConfluentUtils.scala

```diff
@@ -56,4 +59,28 @@ object ConfluentUtils extends LazyLogging {
       .parsePayloadToByteBuffer(bytes)
       .valueOr(exc => throw new SerializationException(exc.getMessage, exc))
       .getInt
+
+  def serializeRecordToBytesArray(record: GenericRecord, schemaId: Int): Array[Byte] = {
```

Comment:
I wonder, can't we use AbstractConfluentKafkaAvroSerializer somehow and not duplicate this low-level stuff?

Reply:
I would like to move in other direction - to stop using `AbstractConfluentKafkaAvroSerializer` and instead of this just use simple methods to serialize/deserialize records. Now its takes too long to figure out how to (de)serialize avro after a while of not being in this subject or for newcomers in this subject. Also currently we already have a part of serializing logic in `ConfluentUtils` - `MagicByte` and `readId`.

Reply:
hmm... oook...